building_sim_plugins/building_plugins_common/include/building_sim_common/crowd_simulator_common.hpp

```diff
@@ -238,132 +246,202 @@ template<typename SdfPtrT>
-
-    if (!model_type_element->template Get<std::string>("filename",
-      model_type_ptr->file_name, ""))
+    if (!sdf->template HasElement("scene_file"))
     {
       RCLCPP_ERROR(logger(),
-        "No actor skin configured in <model_type>! <filename> Required");
+        "No scene file found! <scene_file> Required!");
       return false;
     }
+    _scene_file =
+      sdf->template GetElementImpl("scene_file")->template Get<std::string>();
 
-    if (!model_type_element->template Get<std::string>("animation",
-      model_type_ptr->animation, ""))
+    if (!sdf->template HasElement("update_time_step"))
     {
       RCLCPP_ERROR(logger(),
-        "No animation configured in <model_type>! <animation> Required");
+        "No update_time_step found! <update_time_step> Required!");
       return false;
     }
+    _sim_time_step =
+      sdf->template GetElementImpl("update_time_step")->template Get<float>();
 
-    if (!model_type_element->template Get<double>("animation_speed",
-      model_type_ptr->animation_speed, 0.0))
+    if (!sdf->template HasElement("model_type"))
     {
-      RCLCPP_ERROR(
-        logger(),
-        "No animation speed configured in <model_type>! <animation_speed> Required");
+      RCLCPP_ERROR(logger(),
+        "No model type for agents found! <model_type> element Required!");
       return false;
     }
+    auto model_type_element = sdf->template GetElementImpl("model_type");
+    while (model_type_element)
+    {
+      std::string s;
+      if (!model_type_element->template Get<std::string>("typename", s, ""))
+      {
+        RCLCPP_ERROR(logger(),
+          "No model type name configured in <model_type>! <typename> Required");
+        return false;
+      }
+
+      auto model_type_ptr = this->_model_type_db_ptr->emplace(s,
+          std::make_shared<ModelTypeDatabase::Record>() ); //unordered_map
+      model_type_ptr->type_name = s;
+
+      if (!model_type_element->template Get<std::string>("animation",
+        model_type_ptr->animation, ""))
+      {
+        RCLCPP_ERROR(logger(),
+          "No animation configured in <model_type>! <animation> Required");
+        return false;
+      }
+
+      if (!model_type_element->template Get<double>("animation_speed",
+        model_type_ptr->animation_speed, 0.0))
+      {
+        RCLCPP_ERROR(
+          logger(),
+          "No animation speed configured in <model_type>! <animation_speed> Required");
+        return false;
+      }
+
+      model_type_element = model_type_element->template GetNextElement(
+        "model_type");
+    }
 
-    if (!model_type_element->template HasElement("initial_pose"))
+    if (!sdf->template HasElement("external_agent"))
     {
       RCLCPP_ERROR(
         logger(),
-        "No model initial pose configured in <model_type>! <initial_pose> Required [" + s +
-        "]");
-      return false;
+        "No external agent provided. <external_agent> is needed with a unique name defined above.");
```

Comment:
I wouldn't say that this is necessarily an error, this just means that it's a simulation with only humans and no robots